Library/Homebrew/resource_auditor.rb

```diff
@@ -108,8 +108,16 @@ def audit_resource_name_matches_pypi_package_name_in_url
       return unless url.match?(%r{^https?://files\.pythonhosted\.org/packages/})
       return if name == owner.name # Skip the top-level package name as we only care about `resource "foo"` blocks.
 
-      url =~ %r{/(?<package_name>[^/]+)-}
-      pypi_package_name = Regexp.last_match(:package_name).to_s.gsub(/[_.]/, "-")
+      if url.end_with? ".whl"
+        path = URI(url).path
+        return unless path.present?
+
+        pypi_package_name, = File.basename(path).split("-", 2)
+      else
+        url =~ %r{/(?<package_name>[^/]+)-}
+        pypi_package_name = Regexp.last_match(:package_name).to_s.gsub(/[_.]/, "-")
+      end
```

Comment:
NB: I'll clean this regexp up with another PR (it can also be a simple string split)